docs/guide/search/extend_solr.md

```diff
@@ -0,0 +1,542 @@
+
+In simpler cases, you can apply one of the built-in visitors that correspond 
+to the aggregation type.
+The example below uses `RangeAggregationVisitor`:
+
+``` yaml
+services:
+    App\Query\Aggregation\Solr\PriorityAggregationVisitor:
+        class: EzSystems\EzPlatformSolrSearchEngine\Query\Common\AggregationVisitor\RangeAggregationVisitor
+        factory: ['@EzSystems\EzPlatformSolrSearchEngine\Query\Common\AggregationVisitor\Factory\ContentFieldAggregationVisitorFactory', 'createRangeAggregationVisitor']
+        arguments:
+            $aggregationClass: 'App\Query\Aggregation\PriorityRangeAggregation'
+            $searchIndexFieldName: 'priority_i'
+        tags:
+            - { name: ezplatform.search.solr.query.content.aggregation_visitor }
+            - { name: ezplatform.search.solr.query.location.aggregation_visitor }
+```
+
+The visitor is created by `SearchFieldAggregationVisitorFactory`.
+You provide it with two arguments:
+
+- The aggregation class in `aggregationClass`
+- The field name in search index in `searchIndexFieldName`
+
+Tag the service with `ezplatform.search.solr.query.location.aggregation_visitor`.
+
+For the result extractor, you can use the built-in `RangeAggregationResultExtractor`
+and provide it with the aggregation class in the `aggregationClass` parameter.
+
+Tag the service with `ezplatform.search.solr.query.location.aggregation_result_extractor`.
+
+``` yaml
+services:
+    App\Query\Aggregation\Solr\PriorityAggregationResultExtractor:
+        class: EzSystems\EzPlatformSolrSearchEngine\ResultExtractor\AggregationResultExtractor\RangeAggregationResultExtractor
+        arguments:
+            $aggregationClass: 'App\Query\Aggregation\PriorityRangeAggregation'
+        tags:
+            - { name: ezplatform.search.solr.query.location.aggregation_result_extractor } 
+```
+
+You can use a different type of aggregation, followed by respective visitor and extractor classes:
+
+- `EzSystems\EzPlatformSolrSearchEngine\Query\Common\AggregationVisitor\StatsAggregationVisitor`
+- `EzSystems\EzPlatformSolrSearchEngine\Query\Common\AggregationVisitor\TermAggregationVisitor`
+- `EzSystems\EzPlatformSolrSearchEngine\ResultExtractor\AggregationResultExtractor\StatsAggregationResultExtractor`
+- `EzSystems\EzPlatformSolrSearchEngine\ResultExtractor\AggregationResultExtractor\TermAggregationResultExtractor`
+
+In a more complex use case, you must create your own visitor and extractor.
+
+### Custom aggregation visitor
+
+The aggregation visitor must implement `EzSystems\EzPlatformSolrSearchEngine\Query\AggregationVisitor`:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\Aggregation;
+use eZ\Publish\API\Repository\Values\Content\Query\Aggregation;
+use EzSystems\EzPlatformSolrSearchEngine\Query\AggregationVisitor;
+
+final class PriorityRangeAggregationVisitor implements AggregationVisitor
+{
+    public function canVisit(Aggregation $aggregation, array $languageFilter): bool
```

Comment:
@alongosz @adamwojs Is it OK to use  $languageFilter here as a parameter if we have not declared the use of
`Ibexa\Platform\Contracts\ElasticSearchEngine\Query\LanguageFilter` ?

Reply:
> @alongosz @adamwojs Is it OK to use $languageFilter here as a parameter if we have not declared the use of
> `Ibexa\Platform\Contracts\ElasticSearchEngine\Query\LanguageFilter` ?

How is `LanguageFilter` related to `array`?

Reply:
Dunno, but parameter name caught my attention.

Reply:
> Dunno, but parameter name that's been used in the code sample caught my attention.

It doesn't seem related.